src/store/mutations.ts

```diff
@@ -53,6 +53,60 @@ const mutations: MutationTree<CalChartState> = {
     const currentSS = getSelectedStuntSheet(state);
     currentSS.addDot(dot);
   },
+  syncDotLabelIndices(
```

Comment:
This is a pretty large function. Maybe this can be broken up into sensible pieces?